move/axelar_gateway/sources/types/weighted_signers.move

```diff
@@ -16,21 +16,63 @@ public struct WeightedSigners has copy, drop, store {
 /// ------
 /// Invalid length of the bytes
 const EInvalidLength: u64 = 0;
+const EInvalidThreshold: u64 = 1;
+/// For when operators have changed, and proof is no longer valid.
+const EInvalidOperators: u64 = 2;
+
+///-----
+/// Macros
+///-----
+public(package) macro fun find(
+    $s: &WeightedSigners,
+    $f: |WeightedSigner| -> bool,
+): Option<WeightedSigner> {
+    // Bind $s to a local variable to avoid using macro parameter directly in a path expression which is not allowed
+    let s = $s;
+    let signers = s.signers();
+
+    signers.find_index!($f).map!(|index| signers[index])
+}
 
 /// -----------------
 /// Package Functions
 /// -----------------
+
+/// Create a new weighted signers from the given data.
 public(package) fun peel(bcs: &mut BCS): WeightedSigners {
     let len = bcs.peel_vec_length();
     assert!(len > 0, EInvalidLength);
 
     WeightedSigners {
         signers: vector::tabulate!(len, |_| weighted_signer::peel(bcs)),
         threshold: bcs.peel_u128(),
-        nonce: bytes32::peel(bcs)
+        nonce: bytes32::peel(bcs),
     }
 }
 
+/// Validates the weighted signers. The following must be true:
+/// 1. The signers are in ascending order by their public key.
+/// 2. The threshold is greater than zero.
+/// 3. The threshold is less than or equal to the total weight of the signers.
+public(package) fun validate(self: &WeightedSigners) {
+    let signers = self.signers();
+    assert!(!vector::is_empty(&signers), EInvalidOperators);
+
+    validate_signers_order(signers);
```

Comment:
```suggestion
    self.validate_signers();
```
move the length check and order check under this method, instead of copying the field data